### ISSUE TITLE
Unable to sign in with Firefox

### ISSUE THREAD

**Issue by quickdudley** [hachyderm/community#420]
### Please describe the issue you are seeing.

This issue is happening on Firefox on Linux, Google Chrome and the Mastodon iOS app are unaffected.

When I try to sign in to Hachyderm on Firefox I get an error saying "Security verification failed. Are you blocking cookies?"

![image](https://github.com/hachyderm/community/assets/5417572/d6c3d28f-95b7-442a-a6b1-c15142b33cc4)

I have tried clearing the cookies and other storage items related to hachyderm.io and they are recreated when I reload the page.

### How long have you noticed this issue?

2 days

### What geographic region are you accessing Hachyderm from?

New Zealand

**Comment by Preskton** [comment]
Hi, there, @quickdudley. I did a quick test to see if I can log in using the latest Firefox version, and I didn't seem to encounter the same issue.

<img width="557" alt="image" src="https://github.com/hachyderm/community/assets/6252405/d54be3d6-e96e-42da-a10e-8557a1166b20">

If you're comfortable sharing a bit more information, we can do a bit of further debugging:

- Operating system version
- Firefox version
- Are you using two-factor authentication, like a Yubikey?
- Do you see the same behavior if you use a private window?

**Comment by imsofi** [comment]
Tested this running on Linux with Firefox 111 including a Yubikey, and for me I see no issues. If you can, try to close all firefox windows, then run `firefox --safe-mode` from your command line, and attempt to sign in?

You may also try to go to `about:preferences#privacy`, click "Manage Data", and then delete the entry for hachyderm.io from there.

Let me know if this helps!

**Comment by quickdudley** [comment]
I'm seeing the same issue both in safe mode and after clearing out everything in "Manage Data"

Firefox version is 113.0, build ID is 20230504192738.

OS is Linux Mint 21.1

**Comment by slamp** [comment]
@quickdudley is this issue still exist ?

**Comment by quickdudley** [comment]
@slamp yes. I also just tried with cross-site cookies enabled which didn't resolve the issue (it did solve a similar issue on another website)

**Comment by tehstu** [comment]
For what it's worth, I encountered this in Android Firefox 114 recently. I had been signed in for weeks but decided to clear my cache entirely and had the same error. I wasn't sure what options I could set to allow sign-in, so used Tusky for a while. However, with Firefox 115 I signed in without issue.

Couldn't do any obvious triage on mobile, but hoping this info helps.

**Comment by soupglasses** [comment]
> slamp yes. I also just tried with cross-site cookies enabled which didn't resolve the issue (it did solve a similar issue on another website)

@quickdudley You are saying that other websites have similar issues as here? Could it be that you may have something enabled (like strict tracking protection)? As i would be curious for any further information.

I personally have been running Firefox Linux 111 to 115 for the length of this issue being open without encountering any issue. I am still interested in helping out, as it is definitely odd to get these errors regardless.

If you can try to log into Hachyderm on a LiveUSB of your preferred distro and see if you can reproduce it there? Thank you.

**Comment by quickdudley** [comment]
@soupglasses there was one other site I couldn't sign into because its sign-in process used cross-site cookies. I was able to sign in to it after enabling them, but not Hachyderm. On Monday I will try to sign in with a fresh VM.

**Comment by slamp** [comment]
@quickdudley do you still have the issue ? If yes do you have it in incognito and without any extension ?
I recently switch from Brave to Firefox and I didn't have issue to log on.

**Comment by quickdudley** [comment]
Hi @slamp, yes I'm still seeing this issue including in incognito with no visible extensions. Yesterday I tried Firefox in a virtual machine running the Linux Mint 21.1 disk image and was able to log in to Hachyderm. Possibly there's something in my Firefox data causing this issue or possibly it's because I've kept Firefox up to date on my actual machine. I will try wiping my Firefox data fairly soon and see what happens.
